new/QualityKitchen.Mobile/ViewModels/IncidentResolutionViewModel.cs

```
@@ -42,6 +43,57 @@ private set
 			}
 		}
 
+		private bool displayFullText = false;
+		private int maxTextLengthToDisplay = 80;
+		public Command DisplayLongDescriptionCommand 
+		{ 
+			get
+			{
+				return new Command(HandleDisplayLongDescription);
+			}
+		}
+
+		private void HandleDisplayLongDescription()
+		{
+			displayFullText = !displayFullText;
+			OnPropertyChanged("LongTaskDescription");
+		}
+
+		public string LongTaskDescription
+		{
+			get 
+			{
+				var text = Incident.RelatedTask.Description;
+
+				// TODO: calculate some optimal chars numbers based on current platoform
+				if (text.Length < maxTextLengthToDisplay || displayFullText)
+				{
+					return text;
+				}
+				else
+				{
+					return text.Substring(0, maxTextLengthToDisplay) + "   ... more";
+				}
+			}
+		}
+
+
+		public string ShiftName
+		{
+			get
+			{
+				if (Incident != null)
+				{
+					var shift = App.Instance.Operation.Shifts.Where(f => f.Id == Incident.RelatedTask.ShiftId).SingleOrDefault();
+					if (shift != null)
```

`Task.ShiftId` on the related task should always be set and never be null actually. We're you running into scenarios where `shift` was null? 

If so we should throw an `InvalidOperationException` which is to be considered a global error.